lua/dependency_assist/api.lua

```diff
@@ -12,12 +12,17 @@ end
 --- @param path string
 --- @param cb function
 --- @return number 'the job id'
-function Api:get(path, cb)
+function Api:get(path, cb, provider_header)
```

Comment:
@lvim-tech could you change this function to take a table instead of args directly. I think this is a good example of how this could end up growing, and I'd rather switch the format sooner rather than later.